sycl/test/atomic_ref/sub.cpp

```diff
@@ -165,22 +168,16 @@ void sub_test(queue q, size_t N) {
 // Floating-point types do not support pre- or post-decrement
 template <> void sub_test<float>(queue q, size_t N) {
   sub_fetch_test<float>(q, N);
-  // CHECK-LLVM: declare dso_local spir_func i32
-  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicLoad
-  // CHECK-LLVM-SAME: (i32 addrspace(1)*, i32, i32)
-  // CHECK-LLVM: declare dso_local spir_func i32
-  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
-  // CHECK-LLVM-SAME: (i32 addrspace(1)*, i32, i32, i32, i32, i32)
+  // CHECK-LLVM: declare dso_local spir_func float
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicFAddEXT
+  // CHECK-LLVM-SAME: (float addrspace(1)*, i32, i32, float)
```

Comment:
Can we make sure that the argument is actually negated?

Reply:
In general, I would expect this to be checked within the actual test run in post-commit.
It basically comes down to how detailed we want the IR checks to be, just like in my earlier question:
> Also, should the IR checks in test/atomic_ref/ be updated to include the emulated checks under the macro defined?

The reason we want to have the existing checks for correct function calls is: "Make sure that not only GPU runs pass, but also that they use the native implementation, and not the emulated version that's known to work". A mere check for a correct declaration seems to fit this purpose. The reasoning for testing everything on the IR level (actual call instructions with correct arguments, including the negated ones; "emulated IR" layout under the emulation macro passed, etc.) would have more of a double-checking nature, and would also have to be modified once all BEs support the native atomics.

Given all that, what would you recommend as the best approach?